File: src/main/java/me/andre111/d20server/scripting/expression/Result.java
package me.andre111.d20server.scripting.expression;

public class Result {
	public final double v;
	public final String s;
	public final boolean cf;
	public final boolean cs;
	
	public Result(double value, String string, boolean hadCriticalFailure, boolean hadCriticalSuccess) {
		this.v = value;
		this.s = string;
		this.cf = hadCriticalFailure;
		this.cs = hadCriticalSuccess;
	}
	
	public double getValue() {
		return v;
	}
	
	public String getString() {
		return s;
	}
}

File: core/common/scripting/result.js
export class Result {
    v;
    s;
    drs;

    constructor(value, string, ...diceRolls) {
        this.v = value;
        this.s = string;
        this.drs = diceRolls.flat();
    }

    getValue() {
        return this.v;
    }

    getString() {
        return this.s;
    }

    getDiceRolls() {
        return this.drs;
    }

    hadCriticalFailure() {
        for(const roll of this.drs) {
            if(roll.cf) return true;
        }
        return false;
    }

    hadCriticalSuccess() {
        for(const roll of this.drs) {
            if(roll.cs) return true;
        }
        return false;
    }
}
